05.wc/wc.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  option, file_names = parse_options
+
+  if file_names.empty?
+    print_counts_for_stdin(option)
+  else
+    count_and_print_files(file_names, option)
+  end
+end
+
+def parse_options
+  opt = OptionParser.new
+  option = {}
+  opt.on('-l') { |l| option[:l] = l }
+  opt.on('-w') { |w| option[:w] = w }
+  opt.on('-c') { |c| option[:c] = c }
+  file_names = opt.parse(ARGV)
+  [option, file_names]
+end
+
+def print_counts_for_stdin(option)
+  file_content = $stdin.read
+  line_count, word_count, byte = count(file_content)
+  print_count(line_count, word_count, byte, option, '')
+end
+
+def count(file_content)
+  line_count = file_content.lines.count
+  word_count = file_content.split.count
+  byte = file_content.size
+  [line_count, word_count, byte]
```

Comment:
```suggestion
  { line_count:, word_count:, byte: }
```

3つの戻り値を返す代わりにハッシュを一つ返すのはどうでしょうか？

上記の記法については以下を参照。
[Ruby 3\.1: ES6風のハッシュリテラル省略記法が追加（翻訳）｜TechRacho by BPS株式会社](https://techracho.bpsinc.jp/hachi8833/2021_11_24/113383)